### ISSUE TITLE
S1124: Modifiers should be declared in the correct order

### ISSUE THREAD

**Issue by Pankraz76** [pmd/pmd#6019]
** ModifierOrder:**

- https://sonarsource.github.io/rspec/#/rspec/S1124
- https://docs.openrewrite.org/recipes/staticanalysis/modifierorder
- https://github.com/spotbugs/spotbugs/pull/3657

**Proposed Category:** One of [Best Practices | Code Style | Design | Documentation | Error Prone | Multithreading | Performance | Security]

**Description:**

**Code Sample:** This should include code, that should be flagged by the rule. If possible, the "correct" code
according to this new rule should also be demonstrated.

```java

```

**Possible Properties:**

*   Should this rule be customizable via properties?

please i must be wrong there must be already some rule for this, right?

spend now some time searching for this.

inspired/suggested by
@hazendaz

**Comment by oowekyala** [comment]
See #5601

**Comment by oowekyala** [comment]
This issue does not follow the template. I think it's easier to just use the PR #5601 as a reference.

**Comment by Pankraz76** [comment]
who needs the template if its kind of clear/done already?

**Comment by oowekyala** [comment]
You got my point: we don't need to fill the template, instead we can just close this issue and use the PR as a reference.

But, in the future, please fill the template when you open an issue. Otherwise just don't open an issue, open a discussion.

**Comment by Pankraz76** [comment]
hehe yes and no sorry i meant the you meant the content of the template assuming you demand for details again but you mean the whole in general. I does not matter but idk why this new feat different from others. Having to reference some issue normally following the fix # pattern like seen in:

- https://github.com/pmd/pmd/pull/5856
- https://github.com/pmd/pmd/pull/5847

Best regards

**Comment by oowekyala** [comment]
The informal rule we follow is that if you make a self-contained improvement that is not controversial, you can just open a PR, and we don't necessarily need a separate issue. But if the change needs some discussion, that should happen in an issue. And if you spot an issue and cannot work on a fix, then open an issue, not a PR. Someone else will eventually work on the fix.

In the cases you linked the issues gave rise to some discussions and had relevant information in the main comment. That makes them valuable to track work. This issue here does not match those criteria.
